wrapanapi/systems/rhevm.py

```diff
@@ -64,23 +66,82 @@ def _get_network(self, network_name):
         try:
             return networks[0]
         except IndexError:
-            raise ItemNotFound('No match for network by "name={}"'.format(network_name))
+            raise NotFoundError('No match for network by "name={}"'.format(network_name))
 
-    def update_nic(self, network_name, nic_name='nic1', interface='VIRTIO'):
-        nic_service = self._get_nic_service(nic_name)
-        nic = nic_service.get()
+    def get_nics(self):
+        return self.api.nics_service().list()
+
+    def _nic_action(self, nic, network_name, interface='VIRTIO', on_boot=True,
+                   vnic_profile=None, nic_service=None, action='add'):
+        """Call an action on nic_service, could be a vmnic or vmnics service
+        example, action 'add' on vmnicsservice, or 'update' on VmNicService
+        currently written for nic actions on the service, though other actions are available
+
+        Args:
+            nic: the Nic object itself, could be existing on the vm or not
+            network_name: string name of the network, also default for vnic_profile name if empty
+            interface: string interface type for pyVmomi, interfaces are resolved to a specific type
```

Comment:
One last nitpick here.. we are in rhevm.py, what's pyVmomi doing here? :)

Reply:
works as designed :P